toffy/rosetta.py

```diff
@@ -491,3 +493,48 @@ def create_rosetta_matrices(default_matrix, save_dir, multipliers, masses=None):
                 mult_matrix.iloc[j, :] = comp_matrix.iloc[j, :] * i
         base_name = os.path.basename(default_matrix).split('.csv')[0]
         mult_matrix.to_csv(os.path.join(save_dir, base_name + '_mult_%s.csv' % (str(i))))
+
+
+def copy_image_files(cohort_name, run_names, rosetta_testing_dir, extracted_imgs_dir,
+                     fov_number=10):
+    """ Creates a new directory for rosetta testing and copies over a random subset of
+        previously extracted images
+    Args:
+        cohort_name (str): name for all combined runs
+        run_names (list): gives names of run folders to retrieve extracted images from
+        rosetta_testing_dir: directory where to create cohort rosetta testing folder
+        extracted_imgs_dir: directory containing images from each run
+        fov_number: number of fovs to use for testing, default 10
+
+    """
+    # path validation
+    validate_paths([rosetta_testing_dir, extracted_imgs_dir], data_prefix=False)
+
+    # validate provided run names
+    for run in run_names:
+        if not os.path.exists(os.path.join(extracted_imgs_dir, run)):
+            raise ValueError(f'{run} is not a valid run name found in {extracted_imgs_dir}')
+
+    # make rosetta testing dir and extracted images subdir
+    cohort_rosetta_dir = os.path.join(rosetta_testing_dir, cohort_name)
+    os.makedirs(os.path.join(cohort_rosetta_dir, 'extracted_images'))
+
+    # determine how many fovs to include from each run
+    fovs_per_run = [fov_number // len(run_names) for run in run_names]
```

Comment:
The math here doesn't work if the user has more runs than FOVs. Maybe instead of # of FOVs, it should be # of fovs per run.

Reply:
Makes sense. Should the default # of fovs per run be lowered to maybe 5 then?

Reply:
Yeah I think so